shell/platform/android/platform_view_android.cc

```diff
@@ -519,6 +520,27 @@ void PlatformViewAndroid::RunFromSource(const std::string& assets_directory,
   fml::jni::DetachFromVM();
 }
 
+size_t PlatformViewAndroid::RegisterExternalTexture(
+    const fml::jni::JavaObjectWeakGlobalRef& surface_texture) {
+  return RegisterTexture(
+      std::make_shared<AndroidExternalTextureGL>(surface_texture));
+}
+
+void PlatformViewAndroid::MarkTextureFrameAvailable(size_t texture_id) {
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::Gpu()->PostTask([this, &latch, texture_id]() {
+    std::shared_ptr<AndroidExternalTextureGL> texture =
+        static_pointer_cast<AndroidExternalTextureGL>(
+            rasterizer_->GetTextureRegistry().GetTexture(texture_id));
+    if (texture) {
+      texture->MarkNewFrameAvailable();
+    }
+    latch.Signal();
+  });
+  latch.Wait();
```

Comment:
It should be safe to remove this wait.

The GPU thread will execute tasks in the order they were queued.  So the texture->MarkNewFrameAvailable task will run before the rasterization of the next frame resulting from the call to PlatformView::MarkTextureFrameAvailable.